tests/unit/test_vtk.py

```diff
@@ -1,9112 +1,390 @@
+    save.write_vtu([("dummy_scalar", dummy_scalar), ("dummy_vector", dummy_vector)])
+
+    # Check that exported vtu files and reference files are the same
+    # NOTE: It is implicitly assumed that Gmsh returns the same grid as
+    # for the reference grid; thus, if this test fails, it should be
+    # rerun with an older version of Gmsh to test for failure due to
+    # external reasons.
+    for appendix in ["2", "constant_2"]:
+        assert _compare_vtu_files(
+            f"{setup.folder}/{setup.file_name}_{appendix}.vtu",
+            f"{setup.folder_reference}/constant_data_test_grid_{appendix}.vtu",
+        )
+
+
+def test_fractures_2d(setup):
+    # Test of the export functionality of FractureNetwork2d.
+
+    # Define network
+    p = np.array([[0, 2, 1, 2, 1], [0, 0, 0, 1, 2]])
+    e = np.array([[0, 2, 3], [1, 3, 4]])
+    domain = {"xmin": -2, "xmax": 3, "ymin": -2, "ymax": 3}
+    network_2d = pp.FractureNetwork2d(p, e, domain)
+
+    # Define data
+    dummy_scalar = np.ones(network_2d.num_frac())
+    dummy_vector = np.ones((3, network_2d.num_frac()))
+    data = {"dummy_scalar": dummy_scalar, "dummy_vector": dummy_vector}
+
+    # Make directory if not existent
+    if not os.path.exists(setup.folder):
+        os.makedirs(setup.folder)
+
+    # Export data
+    network_2d.to_file(
+        setup.folder + setup.file_name + ".vtu",
+        data=data,
+    )
+
+    # Check that exported vtu file and reference file are the same
+    # NOTE: It is implicitly assumed that Gmsh returns the same grid as
+    # for the reference grid; thus, if this test fails, it should be
+    # rerun with an older version of Gmsh to test for failure due to
+    # external reasons.
+    assert _compare_vtu_files(
+        f"{setup.folder}/{setup.file_name}.vtu",
+        f"{setup.folder_reference}/fractures_2d.vtu",
+    )
+
+
+def test_fractures_3d(setup):
+    # Test of the export functionality of FractureNetwork3d.
+
+    # Define network
+    f_1 = pp.PlaneFracture(np.array([[0, 1, 2, 0], [0, 0, 1, 1], [0, 0, 1, 1]]))
+    f_2 = pp.PlaneFracture(
+        np.array([[0.5, 0.5, 0.5, 0.5], [-1, 2, 2, -1], [-1, -1, 2, 2]])
+    )
+    domain = {"xmin": -2, "xmax": 3, "ymin": -2, "ymax": 3, "zmin": -3, "zmax": 3}
+    network_3d = pp.FractureNetwork3d([f_1, f_2], domain=domain)
+
+    # Define data
+    num_frac = len(network_3d._fractures)
+    dummy_scalar = [[1] for _ in range(num_frac)]
+    dummy_vector = [[np.ones(3)] for _ in range(num_frac)]
+    data = {"dummy_scalar": dummy_scalar, "dummy_vector": dummy_vector}
+
+    # Make directory if not existent
+    if not os.path.exists(setup.folder):
+        os.makedirs(setup.folder)
+
+    # Export data
+    network_3d.to_file(
+        setup.folder + setup.file_name + ".vtu",
+        data=data,
+    )
+
+    # Check that exported vtu file and reference file are the same
+    # NOTE: It is implicitly assumed that Gmsh returns the same grid as
```

Comment:
Perhaps move this NOTE inside _compare_vtu_files instead of repeating it each time the function is called?

Reply:
I think I disagree with this - I like having such messages to the future self right where they will be needed, in this case next to the place the test will fail.

Reply:
I thought about that (and half expected a disagreement). But isn't the obvious place to check if that assertion breaks inside the function being called? I mean, it shouldn't be difficult to find the note inside the function in this case. Just a suggestion, though, no strong opinions

Reply:
Maybe you are right, but if we both agree, I suggest we leave it to @jwboth to decide.

Reply:
Sure!

Reply:
No super strong feeling. But in the end, I actually also lean towards mentioning it only once in the auxiliary method.